tools/blocksconvert/scanner/cassandra_index_reader.go

```diff
@@ -0,0 +1,157 @@
+package scanner
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+	"github.com/cortexproject/cortex/pkg/chunk/cassandra"
+)
+
+const nbTokenRanges = 512
```

Comment:
Why is this a fixed number, rather than being related to the number of processors?

Reply:
Thank you, added some comments for the mystery value. Cassandra doesn't like having the entire table scanned, so I split it up in many chunks.

Reply:
> Cassandra doesn't like having the entire table scanned, so I split it up in many chunks.

Interesting. I've recently written code scanning entire Cassandra table with hundreds of millions of entries, and haven't run into any issues. Select was very simple: `SELECT <columns> FROM <table>`, with no other conditions. It took a while to process all the results (as expected), but it worked.